src/source-map.ts

```diff
@@ -60,10 +168,51 @@ export class SourceMapGenerator {
     setSourceContent(this._map, source, content);
   }
 
+  applySourceMap(
+    sourceMapConsumer: SourceMapConsumer,
+    sourceFile?: string,
+    sourceMapPath?: string,
+  ) {
+    // If the SourceMap has no file property, we will use `sourceFile`
+    if (sourceMapConsumer.file == null) {
+      if (sourceFile == null) {
+        throw new Error(
+          'SourceMapGenerator.prototype.applySourceMap requires either an explicit source file, ' +
+            'or the source map\'s "file" property. Both were omitted.',
+        );
+      }
+      sourceMapConsumer.file = sourceFile;
+    }
+
+    const sourceRoot = this._map.sourceRoot;
+
+    const remapped = remapping(
+      [toDecodedMap(this._map) as SourceMapInput, sourceMapConsumer],
+      (_, ctx) => {
+        if (sourceMapPath || sourceRoot) {
+          if (sourceMapPath != null) {
+            ctx.source = join(sourceMapPath, ctx.source);
+          }
+
+          if (sourceRoot != null) {
+            ctx.source = relative(sourceRoot, ctx.source);
+          }
+        }
+      },
+      { decodedMappings: true },
+    );
```

Comment:
This seems to fail in quite a few cases
* `sourceRoot` is not properly updated, as it seems to not be handled at all  : https://github.com/ampproject/remapping/blob/main/src/source-map-tree.ts#L85-L86
* It breaks with `Error: Transformation map 0 must have exactly one source file.` Which currently fails in the unit tests

Also, should this be moved to `gen-mapping` ?

Reply:
Resolved as this code is no longer in this PR